src/core/ddsc/src/dds_entity.c

```diff
@@ -1007,6 +1012,12 @@ static void pushdown_listener (dds_entity *e)
 
       ddsrt_mutex_lock (&e->m_observers_lock);
```

Comment:
This line (or, perhaps more precisely, the fact that it is held here until after the `dds_entity_deriver_invoke_cbs_for_pending_events` means in this one path it holds the parent entity's `m_observers_lock` while invoking the listener. I'm sure that'll be fine in many cases, but before I can merge it I want to understand what the effect of that will be.

Or it needs to be released before invoking the listeners. The only thing it is doing for which it requires it is the `dds_override_inherited_listener` call. So moving the "unlock" is probably the best option.

Reply:
In the next pull-request we will unlock m_observers_lock prior to invoking the `dds_entity_deriver_invoke_cbs_for_pending_events`.